botocore/exceptions.py

```diff
@@ -52,6 +53,16 @@ class EndpointConnectionError(BotoCoreError):
         'Could not connect to the endpoint URL: "{endpoint_url}"')
 
 
+class BadStatusLineError(ConnectionError):
+    fmt = (
+        'Bad status line occurred on endpoint URL: "{endpoint_url}". '
```

Comment:
I think "lost" is confusing, it's not something I commonly hear with regard to connections.

What are the *specific* conditions in which we would get an empty string for the status line and not have it be because the server/remote-end closed the connection?

I'm not aware of when else this would happen.